File: apps/mobile/lib/features/organizations/data/models/search_organizations_input.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:altrupets/features/organizations/data/models/organization.dart';

part 'search_organizations_input.freezed.dart';
part 'search_organizations_input.g.dart';

@freezed
class SearchOrganizationsInput with _$SearchOrganizationsInput {
  const factory SearchOrganizationsInput({
    String? name,
    OrganizationType? type,
    OrganizationStatus? status,
    String? country,
    String? province,
    String? canton,
  }) = _SearchOrganizationsInput;

  factory SearchOrganizationsInput.fromJson(Map<String, dynamic> json) =>
      _$SearchOrganizationsInputFromJson(json);
}

File: apps/backend/src/organizations/dto/search-organizations.input.ts
import { InputType, Field } from '@nestjs/graphql';
import { IsString, IsOptional, IsEnum } from 'class-validator';
import { OrganizationType, OrganizationStatus } from '../entities/organization.entity';

@InputType()
export class SearchOrganizationsInput {
  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  name?: string;

  @Field(() => String, { nullable: true })
  @IsEnum(OrganizationType)
  @IsOptional()
  type?: OrganizationType;

  @Field(() => String, { nullable: true })
  @IsEnum(OrganizationStatus)
  @IsOptional()
  status?: OrganizationStatus;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  country?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  province?: string;

  @Field({ nullable: true })
  @IsString()
  @IsOptional()
  canton?: string;
}
